cmd/clusterctl/client/cluster/upgrader.go

```diff
@@ -299,8 +305,8 @@ func (u *providerUpgrader) createCustomPlan(ctx context.Context, upgradeItems []
 			return nil, err
 		}
 
-		if contract != targetContract {
-			return nil, errors.Errorf("unable to complete that upgrade: the provider %s supports the %s API Version of Cluster API (contract), while the management cluster is being updated to %s. Please include the %[1]s provider in the upgrade", provider.InstanceName(), contract, targetContract)
+		if !compatibleContracts.Has(contract) {
+			return nil, errors.Errorf("unable to perform upgrade: the provider %s implements the %s contract version, while the core provider to a version that supports %s contract version. Please include the %[1]s provider in the upgrade", provider.InstanceName(), contract, strings.Join(compatibleContracts.UnsortedList(), ", "))
```

Comment:
> while the core provider to a version that supports %s contract version

I think there's something wrong in this sentence. Not sure how to interpret it

Reply:
Fixed, PTAL

Reply:
works, thx!